Archive Carousel for desktop and mobile

rapterjet2004 opened the pull request:
Fixes #73 
Fixes #74 

Sorry for the late PR, had midterms all of last week. Decided to knock out two birds with one stone on this one. 

- There is a carousel that rotates through images, titles, descriptions, and links every 5 seconds
- The following pictures/description/titles/link are available
- It should span the width of the page
- The user should be able to click through them
- It should use proper font sizing, coloring
- The background image should be darkened
- The images should be locally available
- Responsive, carousel css adjusts for mobile devices (I made sure to use the same bounds as tailwindcss, which is 640px width to indicate small devices. I didn't include anything for tablet screens b/c the format looked alright as is)
- Scalable, to add a new slide, simply add a new image to /assets/images/archiveImages, and then modify the list in ArchiveCarousel/allarchive.jsx to match and the code does the rest
- I also downloaded two (free and transparent) chevron arrow icons for the carousel and included them under assets/icons/
- Added the Carousel to the about and social pages
- Also went ahead and made the WhoWeAreSection a component. 

**Note**: 
Also, I know that the carousel's formatting isn't correct on mobile, but I don't believe that's an error with my code as it effects all components in the about page. I did a quick test of #78 and I believe it fixes the error.

rapterjet2004 commented on src/components/Description/index.jsx:
```diff
@@ -1,7 +1,7 @@
 export default function Description(){
     return (<div className={"container m-auto px-[5rem]"}>
-                    <div className={"mx-[5rem] min-w-full"}>
+                    <div className={"text-center sm:mx-[5rem] sm:min-w-full sm:text-left"}>
```
Don't know why this is showing up as changed, I reverted my prior commit, this should be unchanged file

ChristianBingman commented on src/pages/about.js:
```diff
@@ -2,20 +2,17 @@ import './about.css'
 import ExecutiveLeaders from '../components/ExecutiveLeaders';
 import Description from '../components/Description';
+import ArchiveCarousel from '../components/ArchiveCarousel';
+import WhoWeAreSection from '../components/WhoAreWeSection';
```
Could you call it WhoAreWe instead of WhoAreWeSection

ChristianBingman replied:
It should be WhoAreWe, much better for consistency.

ChristianBingman commented on src/components/ArchiveCarousel/index.jsx:
```diff
@@ -0,0 +1,61 @@
+
+  return (
+    <div className="carousel" >
+      <h1 class="grid place-items-center">Archive</h1>
```
Class is invalid JSX, use classname

ChristianBingman commented on src/components/ArchiveCarousel/index.jsx:
```diff
@@ -0,0 +1,61 @@
+  return (
+    <div className="carousel" >
+      <h1 class="grid place-items-center">Archive</h1>
+      <button className="indicatorLeft" onClick={() => {
```
Could you remove the indicators on mobile?

ChristianBingman commented on src/components/ArchiveCarousel/archivecarousel.css:
```diff
@@ -0,0 +1,29 @@
+.carousel{
```
Most of these can be implemented in tailwind

ChristianBingman commented on src/components/ArchiveCarousel/Item/index.jsx:
```diff
@@ -0,0 +1,26 @@
+
+    return (
+      <div  class="carousel-item" style={{width: "100%"}}>
+        {matches && (<h1 class="title" >{props.title}</h1>)}
```
class is invalid jsx

ChristianBingman commented on src/components/ArchiveCarousel/Item/item.css:
```diff
@@ -0,0 +1,62 @@
+    position: absolute;
+    font-size: .6rem;
+    top: 75%;
+    color: yellow;
```
The link color is secondary dark iirc

ChristianBingman reviewed (changes requested):
Needs a little more changes, few major notes, class is invalid in react since class is reserved in JS, but you are writing inline HTML essentially. I will be upping the estimate on this, I really appreciate all the work on this!

rapterjet2004 commented:
Got it, I
- deleted the css files and used tailwind instead
- changed the website link color to secondary dark
- changed all class keywords to className
- renamed WhoAreWeSection to just WhoAreWe

ChristianBingman reviewed (changes requested):
Very very close, could you fix the naming of WhoAreWe for consistency, and there was a new error in the [build](https://github.com/acm-uic/flourish-2023/actions/runs/3401983003/jobs/5657440283#step:3:43) (You can ignore the prop-type errors for now.

ChristianBingman commented on src/components/ArchiveCarousel/index.jsx:
```diff
@@ -46,7 +46,7 @@ export default function ArchiveCarousel() {
         {
           ARCHIVE.map((c) => {
-            return <Item name={c[0]} title={c[1]} description={c[2]} link={c[3]}/>
+            return <Item key={c} name={c[0]} title={c[1]} description={c[2]} link={c[3]}/>
```
Not a great key, but that'll work

ChristianBingman reviewed (approved):
Great thanks!
